Source/Core/Core/PowerPC/CachedInterpreter/CachedInterpreterEmitter.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/PowerPC/CachedInterpreter/CachedInterpreterEmitter.h"
+
+#include <algorithm>
+#include <cstring>
+
+#include "Common/Assert.h"
+#include "Common/MsgHandler.h"
+
+void CachedInterpreterEmitter::Write(AnyCallback callback, const void* operands, std::size_t size)
+{
+  DEBUG_ASSERT(reinterpret_cast<std::uintptr_t>(m_code) % alignof(AnyCallback) == 0);
+  if (m_code + sizeof(callback) + size >= m_code_end)
+  {
+    m_write_failed = true;
+    return;
+  }
+  std::memcpy(m_code, &callback, sizeof(callback));
+  m_code += sizeof(callback);
+  std::memcpy(m_code, operands, size);
+  m_code += size;
+}
+
+s32 CachedInterpreterEmitter::PoisonCallback(PowerPC::PowerPCState& ppc_state, const void* operands)
+{
+  PanicAlertFmtT("The Cached Interpreter reached a poisoned callback. This should never happen!");
+  return 0;
+}
+
+void CachedInterpreterCodeBlock::PoisonMemory()
+{
+  DEBUG_ASSERT(reinterpret_cast<std::uintptr_t>(region) % alignof(AnyCallback) == 0);
+  DEBUG_ASSERT(region_size % sizeof(AnyCallback) == 0);
+  std::fill(reinterpret_cast<AnyCallback*>(region),
+            reinterpret_cast<AnyCallback*>(region + region_size), AnyCallbackCast(PoisonCallback));
```

Comment:
What happens if `region_size` isn't aligned? Will this terminate?

(I guess in practice it's always aligned, but this smells a bit fishy.)

Reply:
I'm confused what you mean? The first debug assert checks if the memory region is `AnyCallback` aligned. The second debug assert checks that the memory region's size is a multiple of the sizeof `AnyCallback`. These two checks I thought ensured the `std::fill` won't be reached without an assertion in the currently impossible scenario of a bad memory region.

Reply:
Oh, yes, I must have overlooked the second assert.